data_structures/queue/monotonic_queue.py

```diff
@@ -0,0 +1,39 @@
+from __future__ import annotations
+
+from .double_ended_queue import Deque
+
+arr = [1, 3, -1, -3, 5, 3, 6, 7]
+window_size = 3
+expect = [3, 3, 5, 5, 6, 7]
+
+
+def max_sliding_window(arr: list[float], window_size: int) -> list[float]:
+    """
+    Given an array of integers nums, there is a sliding window of size k which is moving
+    from the very left of the array to the very right.
+    Each time the sliding window of length window_size moves right by one position.
+    Return the max sliding window.
+    >>> max_sliding_window(arr, window_size) == expect
+    True
+    """
+    max_val = []
+    queue = Deque()
+    for i in range(len(arr)):
+        # pop the element if the index is outside the window size k
+        if queue and i - queue._front.val >= window_size:
```

Comment:
Using the _private_ `_front` is fairly risky.  I.e. Not future-proof.
```

Reply:
Yes! I use the collections.deque now